hal/ia32/memory.c

```diff
@@ -66,51 +79,141 @@ int hal_memoryGetEntry(unsigned int *offs, mem_entry_t *entry)
 		"1: "
 		"addl $0xc, %%esp; "
 	: "+a" (ret), "+b" (next)
-	: "D" (entry)
+	: "D" (&mem)
 	: "ecx", "edx", "memory", "cc");
+
 	*offs = next;
 
+	entry->start = mem.addr;
+	/* TODO: mapent_t should use addr & len instead of start & end */
+	entry->end = (mem.addr + mem.len <= 0xffffffff) ? mem.len + mem.addr : mem.addr + mem.len - 1;
+	entry->type = hal_entryReserved;
+
+	*biosType = mem.type;
```

Comment:
Yeah, you have right, that after fail of interrupt the wrong data will be write to the `entry`. 
However, the function will return error anyway, so this data will not taken into account. So that, I leave it as it is.